### PR TITLE
ATO-875: FIx build and deploy from dev pipeline in GHA

### PR DESCRIPTION
**Pull request by isaac-GDS**
switched to npm to run build
added permission boundary to template
added signing config to template

### CODE REVIEW

File: .github/workflows/deploy-stubs.yml [hunk 1]
```diff
@@ -44,13 +44,16 @@ jobs:
         with:
           path: .aws-sam/cache
           key: orch-sam
+      
+      - name: Install dependencies
+        run: npm install
```
Comment (billrdunn) [inline .github/workflows/deploy-stubs.yml:49]: I think `npm ci` is more appropriate here than `npm install`, see https://stackoverflow.com/questions/52499617/what-is-the-difference-between-npm-install-and-npm-ci

**Comment by billrdunn** [comment]
Can we use some sort of caching like is done here?
https://github.com/govuk-one-login/authentication-api/commit/891141c34800d88e54f4e5b1f89187c6d200bee2

**Comment by isaac-GDS** [comment]
> Can we use some sort of caching like is done here? [govuk-one-login/authentication-api@891141c](https://github.com/govuk-one-login/authentication-api/commit/891141c34800d88e54f4e5b1f89187c6d200bee2)

We have caching at stages before https://github.com/govuk-one-login/orch-stubs/blob/main/.github/workflows/run-checks.yml . Good enough for now ?

File: .github/workflows/deploy-stubs.yml (hunk 1 above)
Comment (isaac-GDS) [reply .github/workflows/deploy-stubs.yml:49]: Will have to look further on this - maybe could be used in another place as it always deletes node_modules. Unsure interaction with caching.

**Review by billrdunn — APPROVED** [review]

**Pull request merged**